tests/common_fixtures.py

```diff
@@ -295,12 +295,12 @@ def assert_s3_mount_configured(self, s3_params):
         env_list = self._function["spec"]["env"]
         param_names = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
         secret_name = s3_params.get("secret_name")
+        non_anonymous = s3_params.get("non_anonymous")
 
-        attr_name = "valueFrom" if secret_name else "value"
         env_dict = {
-            item["name"]: item[attr_name]
+            item["name"]: item["valueFrom"] if "valueFrom" in item else item["value"]
             for item in env_list
-            if item["name"] in param_names
+            if item["name"] in param_names + ["S3_NON_ANONYMOUS"]
```

Comment:
why you chose to separate `S3_NON_ANONYMOUS` from param_names?

Reply:
Because I later iterate over `param_names` to generate the expected results, and if the test used a secret then it generates everything with a `secretKeyRef`. However, the `S3_NON_ANONYMOUS` parameter is never retrieved from a secret so I need to generate the expected result differently. So, I chose to do it like this - this entire function is a bit ugly because of the various options that exist... Still - it's a test, so I didn't see much point in investing too much in refactoring it.